components/feature/search/src/main/java/mozilla/components/feature/search/middleware/SearchMiddleware.kt

```diff
@@ -243,4 +296,28 @@ class SearchMiddleware(
             val searchEngineName: String?
         )
     }
+
+    /**
+     * Interface for a class that can provide data from a legacy system to be imported into the
+     * storage used by the middleware.
+     */
+    interface Migration {
+        /**
+         * Returns the values to be migrated. It is expected that the application returns the values
+         * only once. Afterwards the data is assumed to be migrated and should not be provided again.
```

Comment:
Do we need a way for the app to know that the migration was successful? Re: `Afterwards the data is assumed to be migrated and should not be provided again.` -> Should the app just try once and assume success or check state in the store?

Reply:
Yeah, I was planning to only try once and not implement a complex migration protocol.